pkg/cluster/config/encoding/scheme.go

```diff
@@ -50,12 +50,28 @@ func AddToScheme(scheme *runtime.Scheme) {
 
 // Load reads the file at path and attempts to convert into a `kind` Config; the file
 // can be one of the different API versions defined in scheme.
-// If path == "" then the default config is returned
-func Load(path string) (*config.Config, error) {
+// If path == "" then the default config is returned according to requested number of
+// control-plane and worker nodes
+func Load(path string, controlPlanes, workers int32) (*config.Config, error) {
```

Comment:
Combining this logic into load seems unusual and possibly hostile to library users.

Reply:
If we have those fields set, we should just create an internal config in the cmd, set these fields, then default.

Reply:
👍 it feels awkward to pass some random extra vars to this function